### PR TITLE
Propose module organization

### PR DESCRIPTION
**Pull request by apeltzer**
See above changes to readme :_)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,6 +14,21 @@ A repository for hosting nextflow [`DSL2`](https://www.nextflow.io/docs/edge/dsl
     * [Uploading to `nf-core/modules`](#uploading-to-nf-coremodules)
 * [Help](#help)
 
+## Terminology
+
+The DSLv2 features for Nextflow are new for everyone and not just beginners. We discussed some terminology terms to discuss things related to modules & subworkflows more appropriately:
+
+* *Module*: An "atomic" module (one thing), e.g. a reusable process from an old pipeline that can be used between different pipelines
```
Comment (mahesh-panchal) [inline README.md]: * *Module*: A `process` that can be used in different workflows, which is atomic (i.e. can/should not be divided further).

File: README.md [hunk 2]
```diff
@@ -14,6 +14,21 @@ A repository for hosting nextflow [`DSL2`](https://www.nextflow.io/docs/edge/dsl
...
+* *Subworkflow*: A combined set of modules, that do e.g. preprocessing for a much bigger pipeline. A good example could be a QC subworkflow for FastQ input, that could be (re-) used across multiple pipelines
```
Comment (mahesh-panchal) [inline README.md]: * *Subworkflow*: A set of modules combined using `workflow <name>`, that performs a logical stage within a bigger pipeline, e.g., a quality control subworkflow for FastQ input.

File: README.md [hunk 3]
```diff
@@ -14,6 +14,19 @@ A repository for hosting nextflow [`DSL2`](https://www.nextflow.io/docs/edge/dsl
...
+
+The DSLv2 features for Nextflow are new for everyone and not just beginners. We discussed some terminology terms to discuss things related to modules & subworkflows more appropriately:
+
+* *Module*: A `process`that can be used between different pipelines, which is atomic (i.e. can/should not be divided further).
+* *Subworkflow*: A combined set of modules, that combine a logical step in a pipeline using multiple modules together. Example: A preprocessing subworkflow for FastQ input, that could be (re-) used across multiple pipelines to QC input FastQ files.
+* *Workflow*: What DSLv1 users would consider a pipeline, e.g. from input to potentially complex output. Can either consist of individual modules, a large monolithic script as in DSLv1 or a combination of subworkflows (or any combination of these three). 
+
+## Organization of repository
+
+* Modules should end up in the subdirectory `tools`
```
Comment (drpatelh) [inline README.md]:
```suggestion
* This repository has been created to only host atomic module files that should be added to the `tools` sub-directory along with the required documentation, software and tests.
```

File: README.md [hunk 4]
```diff
@@ -14,6 +14,19 @@ A repository for hosting nextflow [`DSL2`](https://www.nextflow.io/docs/edge/dsl
...
+* Subworkflows should be kept with individual pipelines, and not end up here in the modules repository.
```
Comment (drpatelh) [inline README.md]:
```suggestion
* Sub-workflows should be shipped with the pipeline implementation and if required they should be shared amongst different pipelines directly from there. As it stands, this repository will not host sub-workflows.
```

File: README.md [hunk 5]
```diff
@@ -14,6 +14,19 @@ A repository for hosting nextflow [`DSL2`](https://www.nextflow.io/docs/edge/dsl
     * [Uploading to `nf-core/modules`](#uploading-to-nf-coremodules)
 * [Help](#help)
 
+## Terminology
+
+The features offered by Nextflow DSL 2 can be used in various ways depending on the granularity with which you would like to write pipelines. Please see the listing below for the hierarchy and associated terminology we have decided to use when referring to DSL 2 components:
+
+* *Module*: A `process`that can be used within different pipelines and is as atomic as possible i.e. cannot be split into another module. An example of this would be a module file containing the process definition for a single tool such as `FastQC`.
+* *Sub-workflow*: A chain of multiple modules that offer a higher-level of functionality within the context of a pipeline.  For example, a sub-workflow to run multiple QC tools with FastQ files as input.
```
Comment (drpatelh) [inline README.md]:
```suggestion
* *Sub-workflow*: A chain of multiple modules that offer a higher-level of functionality within the context of a pipeline.  For example, a sub-workflow to run multiple QC tools with FastQ files as input. Sub-workflows should be shipped with the pipeline implementation and if required they should be shared amongst different pipelines directly from there. As it stands, this repository will not host sub-workflows.
```

File: README.md [hunk 6]
```diff
@@ -14,6 +14,19 @@ A repository for hosting nextflow [`DSL2`](https://www.nextflow.io/docs/edge/dsl
...
+* *Workflow*: What DSL 1 users would consider an end-to-end pipeline. For example, from one or more inputs to a series of outputs. This can either be implemented using a large monolithic script as with DSL 1, or by using a combination of DSL 2 individual modules and sub-workflows. 
+
+## Organization of repository
+
+* This repository has been created to only host atomic module files that should be added to the `tools` sub-directory along with the required documentation, software and tests.
+* Sub-workflows should be shipped with the pipeline implementation and if required they should be shared amongst different pipelines directly from there. As it stands, this repository will not host sub-workflows.
+
```
Comment (drpatelh) [inline README.md]:
```suggestion
```

**Comment by ewels** [comment]
Nice! 👍🏻

**Pull request merged**